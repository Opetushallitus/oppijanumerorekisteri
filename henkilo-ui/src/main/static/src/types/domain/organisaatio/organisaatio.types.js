// @flow

export type OrganisaatioHakuTulos = {
    numHits: number,
    organisaatiot: Array<Organisaatio>
}

export type Organisaatio = {
    oid: string,
    alkuPvm: string,
    lakkautusPvm: string,
    parentOid: string,
    parentOidPath: string,
    ytunnus: string,
    oppilaitosKoodi: string,
    oppilaitostyyppi: string,
    toimipistekoodi: string,
    match: boolean,
    nimi: {
        fi?: string,
        sv?: string,
        en?: string
    },
    kieletUris: Array<string>,
    kotipaikkaUri: string,
    children: Array<Organisaatio>,
    aliOrganisaatioMaara: number,
    virastoTunnus: string,
    organisaatiotyypit: Array<string>,
    status: string
}

export type Ryhma = {
    oid: string,
    nimi: {[string]: string},
    version: number,
    parentOid: string,
    parentOidPath: string,
    alkuPvm: string,
    lakkautusPvm: string,
    kuvaus: {[string]: string},
    ryhmatyypit: Array<string>,
    kayttoryhmat: Array<string>
}