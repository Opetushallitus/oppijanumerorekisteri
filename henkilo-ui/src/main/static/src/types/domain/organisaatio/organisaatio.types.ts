export type OrganisaatioStatus = 'AKTIIVINEN' | 'SUUNNITELTU' | 'PASSIIVINEN' | 'POISTETTU';

export type Organisaatio = {
    oid: string;
    parentOidPath: string;
    nimi: {
        fi?: string;
        sv?: string;
        en?: string;
    };
    tyypit: Array<string>;
    status: OrganisaatioStatus;
};

export type OrganisaatioWithChildren = {
    oid: string;
    parentOidPath: string;
    nimi: {
        fi?: string;
        sv?: string;
        en?: string;
    };
    tyypit: Array<string>;
    children: Array<OrganisaatioWithChildren>;
    status: OrganisaatioStatus;

    // generated by ui
    parentOid: string | null | undefined;
    parent: OrganisaatioWithChildren;
};

export type OrganisaatioTyyppi = 'ORGANISAATIO' | 'RYHMA';

export type OrganisaatioCriteria = {
    tyyppi?: OrganisaatioTyyppi;
    tila?: Array<OrganisaatioStatus>;
};
