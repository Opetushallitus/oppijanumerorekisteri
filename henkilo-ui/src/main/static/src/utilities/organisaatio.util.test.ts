import { omattiedotOrganisaatiotToOrganisaatioSelectObject } from './organisaatio.util';
import type { OrganisaatioWithChildren } from '../types/domain/organisaatio/organisaatio.types';

describe('omattiedotOrganisaatiotToOrganisaatioSelectObject', () => {
    const parent: OrganisaatioWithChildren = {
        oid: '2',
        parentOid: '1',
        parentOidPath: '1',
        nimi: {
            fi: 'b',
        },
        status: 'AKTIIVINEN',
        tyypit: [],
        parent: null,
        children: [],
    };

    const child: OrganisaatioWithChildren = {
        oid: '3',
        parentOid: '2',
        parentOidPath: '1/2',
        nimi: {
            fi: 'c',
        },
        status: 'AKTIIVINEN',
        tyypit: [],
        parent: null,
        children: [],
    };

    parent.children = [child];
    child.parent = parent;

    test('handles empty input', () => {
        expect(omattiedotOrganisaatiotToOrganisaatioSelectObject([], 'fi')).toMatchObject([]);
    });

    test('resolve organisation hierarchy correctly', () => {
        expect(omattiedotOrganisaatiotToOrganisaatioSelectObject([{ organisaatio: parent }], 'fi')).toMatchSnapshot();
    });
});
