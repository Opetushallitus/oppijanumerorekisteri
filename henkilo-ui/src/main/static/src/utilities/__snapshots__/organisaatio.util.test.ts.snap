// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omattiedotOrganisaatiotToOrganisaatioSelectObject resolve organisation hierarchy correctly 1`] = `
Array [
  Object {
    "name": "b",
    "oid": "2",
    "oidPath": "1",
    "organisaatiotyypit": Array [],
    "parentNames": Array [],
    "status": "AKTIIVINEN",
  },
  Object {
    "name": "c",
    "oid": "3",
    "oidPath": "1/2",
    "organisaatiotyypit": Array [],
    "parentNames": Array [
      "b",
    ],
    "status": "AKTIIVINEN",
  },
]
`;
